fireapp/lib/presentation/shift_request/shift_request_view_model.dart

```diff
@@ -28,18 +31,32 @@ class ShiftRequestViewModel extends FireAppViewModel {
 
   ShiftRequestViewModel(this._shiftRequestRepository);
 
-  void loadShiftRequests(String requestId) {
-    _shiftRequests.add(RequestState.loading());
-    () async {
-      try {
-        final requests = await _shiftRequestRepository.getShiftRequestsByRequestID(requestId);
-        _shiftRequests.add(RequestState.success(requests));
-      } catch (e) {
-        logger.e(e);
-        _shiftRequests.add(RequestState.exception(e));
+  void loadShiftRequests({String? requestId, bool useMockData = false}) async {
+  _shiftRequests.add(RequestState.loading());
+
+  try {
```

Comment:
This shouldn't be necessary. The only thing that should be caring about whether you are loading mock data is the client. The client should be responsible for loading, deserialising and returning the mock data.